syslog/src/syslogsink.cpp

```diff
@@ -20,12 +20,27 @@ void SyslogSink::echoToStderr()
     _option |= LOG_PERROR;
 }
 
+void SyslogSink::setIdentity(const char* id)
+{
+// https://www.gnu.org/software/libc/manual/html_node/openlog.html
+// syslog doesn't copy the identity string internally, it only keeps a pointer to the string.
+// We have to create a new string, then switch the pointer used by syslog, then
+// delete the previous string.
+// note that we have only one element in the list normally,
+// but here we need to insert a second one temporarily, as 
+// syslog's internal pointer must never point to an invalid memory location.
+_identity_list.push_front(std::string(id));
+::openlog(_identity_list.begin() -> c_str(), _option, _facility);
+// at this point syslog has a new valid string, we can remove the previous one:
+_identity_list.pop_back();
```

Comment:
Ouch. I never know that openlog was so gnarly and thread-unsafe 
https://www.gnu.org/software/libc/manual/html_node/openlog.html
```
Please note that the string pointer ident will be retained internally by the Syslog routines. You must not free the memory that ident points to. It is also dangerous to pass a reference to an automatic variable since leaving the scope would mean ending the lifetime of the variable. If you want to change the ident string, you must call openlog again; overwriting the string pointed to by ident is not thread-safe.
```

Reply:
I probably would have used a `std::unique_ptr<string>` and done a `std::swap` instead. I think that’s clearer logic but this works fine

Reply:
indeed much clearer! I change that.